### PR TITLE
Create a factory for the ActorStack

### PR DESCRIPTION
**Pull request by jcoyne**
@projecthydra/sufia-code-reviewers

This enables Sufia to customize the ActorStack slightly.

Fixes projecthydra/sufia#1776

Will be used to complete work on:
Ref projecthydra/sufia#1699

### CODE REVIEW

File: app/actors/curation_concerns/actor_stack.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module CurationConcerns
+  class ActorStack
+    attr_reader :curation_concern, :user, :first_actor_class, :more_actors
+    def initialize(curation_concern, user, more_actors)
+      @curation_concern = curation_concern
+      @user = user
+      @more_actors = more_actors
+      @first_actor_class = @more_actors.shift || RootActor
```
Comment (cbeer) [inline app/actors/curation_concerns/actor_stack.rb]: What's the difference between `RootActor` here and `BaseActor` below?

Comment (jcoyne) [inline app/actors/curation_concerns/actor_stack.rb]: BaseActor is the super class for all *WorkActors.  RootActor is the terminator of the stack, it's follows the null object pattern.

Comment (cbeer) [inline app/actors/curation_concerns/actor_stack.rb]: Got it. I wonder if we can give those things better names some time.

Comment (jcoyne) [inline app/actors/curation_concerns/actor_stack.rb]: I'm open to suggestions.

Comment (mjgiarlo) [inline app/actors/curation_concerns/actor_stack.rb]: `NullActor`?

Comment (jcoyne) [inline app/actors/curation_concerns/actor_stack.rb]: @mjgiarlo I considered that, but the RootActor returns true for all methods, so calling it the NullActor seems confusing.

Comment (jcoyne) [inline app/actors/curation_concerns/actor_stack.rb]: RecursiveBaseCaseActor?

Comment (jcoyne) [inline app/actors/curation_concerns/actor_stack.rb]: In any case, neither of these classes were created in this PR, so if you've got a good name please open a ticket, or better yet, a PR.

Comment (mjgiarlo) [inline app/actors/curation_concerns/actor_stack.rb]: Yeah, I wouldn't hold up the PR over the name. :shipit:

Comment (cbeer) [inline app/actors/curation_concerns/actor_stack.rb]: Agreed.

**Pull request merged**